### PR TITLE
985 add work type field to time entry dialog

### PR DESCRIPTION
**Pull request by kti-cameron**

### CODE REVIEW

**Comment by kti-sam** [comment]
You need to run a migration merge on this still, the migrations still conflict

**Comment by kti-cameron** [comment]
Ok. Should be fixed now.

**Comment by kti-sam** [comment]
How exactly did you do the migration merge? I don't see a merge migration and looking here now and it looks like there is a migration 0088 on master, and a different 0088 on #222.

**Comment by kti-sam** [comment]
Oh I just saw the deleted file. You can not delete a migration like that. 0088_auto_20190710_0953.py is already on the master branch, the database is migrated.

Check `makemigrations.py`, line 18. There is a note explaining what to do when you need to merge django-connectwise migrations. For now please restore `0088_auto_20190710_0953`

**Comment by kti-sam** [comment]
A better explanation for all the reasons that is bad practice would probably be good. 

In this case I think only we have this version, but other times migrations may already be on client instances. Depending on the migration we can't or really shouldn't roll back migrations. Not to mention we don't really have a process for that (My info could be out of date, but I don't think that has changed), and we would need a VERY good reason to do that. 

Also other developers might already be working on issues that depend on those migrations, we are a small team and I already have a branch that depends on 0088-0953. Imaging a team of 5 or more devs. 

Basically assume that a migration is set in stone once its been pushed from your local repe. Delete all you want until that MR has been put through, but after that its best to just push a new migration to fix the old one. Obviously there are some exceptions, but we will deal with those as they come.

**Comment by kti-cameron** [comment]
Thanks for the clarification and explanation. I think I understand migration merges better now. I totally missed the fact that `0088_auto_20190710_0953.py` was already on master. 

I ran a revert to restore the deleted migration, then ran the migration merge. I also found that I missed making a migration for modifying work type field, so added that too. 

Hopefully that'll clean up this mess.
